### PR TITLE
Template filtering

### PR DESCRIPTION
**Pull request by thecalcc**
SDESK-7084
SDESK-7085

### CODE REVIEW

File: scripts/api/user.ts [hunk 1]
```diff
@@ -6,6 +6,12 @@ function hasPrivilege(privilege: string): boolean {
     return privileges.userHasPrivileges({[privilege]: 1});
 }
 
+function getCurrentUserId(): string {
```
Comment (tomaskikutis) [inline scripts/api/user.ts:9]: reuse this one https://github.com/superdesk/superdesk-client-core/blob/authoring-react-post-broadcasting/scripts/core/superdesk-api.d.ts#L2861

in the implementation it's inlined, so extract into a separate function

File: scripts/core/ui/components/content-create-dropdown/more-templates.tsx [hunk 2]
```diff
@@ -28,18 +29,34 @@ export class MoreTemplates extends React.PureComponent<IProps, IState> {
...
             method: 'GET',
             path: '/content_templates',
             urlParams: {
                 max_results: pageSize,
                 page: pageToFetch,
-                sort: nameof<ITemplate>('template_name'),
-                where: this.state.searchString.length < 1 ? undefined : {
-                    [nameof<ITemplate>('template_name')]: {
+                sort: templateName,
+                where: this.state.searchString.length < 1 ? {$and: [criteria]} : {
```
Comment (tomaskikutis) [inline scripts/core/ui/components/content-create-dropdown/more-templates.tsx:54]:
```
const where = {$and: [criteria]};

if (this.state.searchString.length < 1) {
    where[templateName] = {
        $regex: this.state.searchString,
        $options: '-i',
    },
}

httpRequestJsonLocal(where, ...)
```

Avoid repeating the same logic twice ( `{$and: [criteria]}` ). In my pseudo-code it's explicit that `{$and: [criteria]}` is always applied and search is applied conditionally. In your version it wouldn't be that clear for someone adding more filters. Should he add it on both places or only one for example? It is clear at the moment, but as code grows it becomes an issue.

File: scripts/api/user.ts (hunk 1 above)
Comment (thecalcc) [reply scripts/api/user.ts:9]: How should I import this?

Comment (tomaskikutis) [reply scripts/api/user.ts:9]: https://github.com/superdesk/superdesk-client-core/blob/authoring-react-post-broadcasting/scripts/core/get-superdesk-api-implementation.tsx#L337

look at implementation of this. First you extract it into a function and then use that function anywhere in the code AND also expose via extensions API

**Review by tomaskikutis — APPROVED** [review]

**Pull request merged**